src/parent/parent.js

```diff
@@ -729,7 +729,16 @@ export function parentComponent<P, X, C>({
         return clean.all(err);
       })
       .then(() => {
-        initPromise.asyncReject(err || new Error("Component destroyed"));
+        const error = err || new Error("Component destroyed");
+        if (
+          isElementClosed(currentContainer) ||
+          error.message === "Window navigated away"
+        ) {
+          console.warn(error);
```

Comment:
I'm leaning towards not logging anything to the console for this use case. If the window is closed or the user navigated away from the page then we can have destroy be silent. 
```suggestion
```